tests/test_upstream_versions.py

```diff
@@ -1643,3 +1643,41 @@ def test_main(
     update_upstream_versions_mock.assert_called_once_with(
         gx, debug=debug, job=3, n_jobs=10, package="testpackage"
     )
+
+
+@pytest.mark.parametrize(
+    "url, version, version_prefix",
+    [
+        ("https://example.com/archs/1.2.3.tar.gz", "1.2.3", None),
+        # we don't want any assumptions on non github urls
+        ("https://example.com/archs/example-1.2.3.tar.gz", "1.2.3", None),
+        (
+            "https://github.com/example/example/archive/refs/tags/1.2.3.tar.gz",
+            "1.2.3",
+            None,
+        ),
+        (
+            "https://github.com/example/example/archive/refs/tags/example-1.2.3.tar.gz",
+            "1.2.3",
+            "example-",
```

Comment:
This isn't always version prefix even for github. Some github urls like this

> https://github.com/simplejson/simplejson/releases/download/{{ version }}/simplejson-{{ version }}.tar.gz

are actually the package name.

Reply:
I think for urls with `/refs/tags` in them, the test case is correct.

Reply:
I've just added support to find first split with version. There still could be examples like:
```
https://github.com/simplejson/simplejson/releases/download/my-name-1.2.3/simplejson-1.2.3.tar.gz
```

And currently it won't find it because of the original issue. Prefix is `my-name-` that is not predefined and contains `a` in name. So it will be treated as alpha version and ignored.